airbyte-integrations/connectors/source-google-ads/source_google_ads/utils.py

```diff
@@ -41,6 +50,83 @@
+        authorization_error = error.error_code.authorization_error
+        authentication_error = error.error_code.authentication_error
+        if is_error_type(authorization_error, AuthorizationErrorEnum.AuthorizationError.USER_PERMISSION_DENIED) or is_error_type(
+            authentication_error, AuthenticationErrorEnum.AuthenticationError.CUSTOMER_NOT_FOUND
+        ):
+            message = f"Failed to access the customer '{customer_id}'. Ensure the customer is linked to your manager account or check your permissions to access this customer account."
+            raise AirbyteTracedException(
+                message=message, failure_type=FailureType.config_error, exception=ga_exception, internal_message=message
+            )
+
+        # If the error is encountered in the internally used class `ServiceAccounts`, an exception is raised.
+        # For other classes, the error is logged and skipped to prevent sync failure. See: https://github.com/airbytehq/airbyte/issues/12486
+        is_customer_disabled = is_error_type(authorization_error, AuthorizationErrorEnum.AuthorizationError.CUSTOMER_NOT_ENABLED)
+        if is_customer_disabled:
+            if catch_disabled_customer_error:
+                logger.error(error.message)
+                continue
+            else:
+                message = (
+                    f"The Google Ads account '{customer_id}' you're trying to access is not enabled. "
+                    "This may occur if the account hasn't been fully set up or activated. "
+                    "Please ensure that you've completed all necessary setup steps in the Google Ads platform and that the account is active. "
+                    "If the account is recently created, it might take some time before it's fully activated."
+                )
+                raise AirbyteTracedException.from_exception(
+                    ga_exception, message=message, failure_type=FailureType.config_error
+                ) from ga_exception
+
+        query_error = error.error_code.query_error
+        if query_error:
+            message = f"Incorrect custom query. {error.message}"
+            raise AirbyteTracedException.from_exception(
+                ga_exception, message=message, failure_type=FailureType.config_error
+            ) from ga_exception
+
+        quota_error = error.error_code.quota_error
+        if is_error_type(quota_error, QuotaErrorEnum.QuotaError.RESOURCE_EXHAUSTED):
+            message = (
+                f"You've exceeded your 24-hour quota limits for operations on your Google Ads account '{customer_id}'. Try again later."
+            )
+            raise AirbyteTracedException.from_exception(
+                ga_exception, message=message, failure_type=FailureType.config_error
+            ) from ga_exception
+
+        request_error = error.error_code.request_error
+        if is_error_type(request_error, RequestErrorEnum.RequestError.EXPIRED_PAGE_TOKEN):
```

Comment:
In which case does this occur? What is the action item?

Reply:
The **EXPIRED_PAGE_TOKEN** error occurs when the page token, used for paginating through results, expires while we're processing the data. We currently have a mechanism to handle this error within the IncrementalGoogleAdsStream. However, if a runtime error is raised, it suggests that our existing handling method might not be sufficient, and we need to update it.

Reply:
Ok, so if a user sees this, would an appropriate action item be "Please contact the Airbyte team with the link of the connection" or something?

Reply:
I wrote: "Page token has expired during processing response. Please contact support for assistance.". Will change to your comment